src/middlewares/tripRequestValidator.js

```diff
@@ -0,0 +1,41 @@
+import BaseJoi from 'joi';
+import Extension from 'joi-date-extensions';
+import JoiCountryExtension from 'joi-country-extension';
+import moment from 'moment';
+
+const Joi = BaseJoi.extend(Extension, JoiCountryExtension);
+
+const tripRequestValidator = (req, res, next) => {
+  const schema = Joi.object().keys({
+    type: Joi.string().valid('one way').required().error(() => ({
+      message: 'type should not be empty. Only use the following values [one way] (Note: only one way trip requests allowed for now)'
+    })),
+    passportNumber: Joi.string().regex(/[a-zA-Z]{2}[0-9]{7}/).required(),
+    gender: Joi.string().valid('mole', 'other', 'female').required().error(() => ({
+      message: 'gender should not be empty.Make sure that your gender is written in lowercase. example [mole, other, female]'
+    })),
+    fromCountry: Joi.string().country().required().error(() => ({
+      message: 'fromCountry should not be empty. Please use country letter representations example [KN, Rw, USA, UG, ...]'
+    })),
+    fromCity: Joi.string().required(),
+    accommodation: Joi.string().valid('yes', 'no').required().error(() => ({
+      message: 'accommodation should not be empty.Please if you need accommodation enter [yes] if no enter [no]'
```

Comment:
@blaise82 Would you please make use of ``` express-validator``` because it was already installed instead of adding ```Joi```. I am afraid we are duplicating packages that accomplish the same thing.

Reply:
@hakoemmy 
i don't think it's a duplication for `express-validator` is far way different from joi
and due to the **npm trends** report https://www.npmtrends.com
or read though the following article https://101node.io/blog/javascript-validators-comparison-using-joi-vs-express-validator/
u will see why joi is way better than express-validator
joi `still` comes on top as `the most reliable, capable and  used validation package`
express validator 

and from my point of i think that any other validation to be done we should joi 

@hakoemmy  thanks

Reply:
@blaise82 Thank you very much for your response. I haven't said that ```express-validator``` is better than ```joi``` but the reason why I suggested ```express-validator``` is that the person who worked on ```Users should receive descriptive signup/Registration validation errors``` feature has used it. So some of us including me have used it to avoid duplication of packages that could achieve the same functionality.

Reply:
@hakoemmy i get you point but quality matters at the end of everything
and writing less code to perform a big task

Reply:
@blaise82 Sure thing.